cmake/usFunctionEmbedResources.cmake

```diff
@@ -138,7 +138,7 @@ function(usFunctionEmbedResources)
       # section name is "us_resources" because max length for section names in Mach-O format is 16 characters.
       add_custom_command(
         OUTPUT ${_source_output}
-        COMMAND ${CMAKE_CXX_COMPILER} -isysroot ${CMAKE_OSX_SYSROOT} -c ${US_CMAKE_RESOURCE_DEPENDENCIES_CPP} -o stub.o
+        COMMAND ${CMAKE_CXX_COMPILER} ${US_OSX_CXX_FLAGS} -c ${US_CMAKE_RESOURCE_DEPENDENCIES_CPP} -o stub.o
```

Comment:
That function is part of our "CMake API", which means it is exposed to clients. As such, the `CMAKE_OSX_CXX_FLAGS` variable will never be set for client projects. I suggest to just move the added CMake block from the top-level CMakeLists.txt file into the usFunctionEmbedResources.cmake file.